### PR TITLE
Account wide Reagent Bank/Bag

### PR DESCRIPTION
**Pull request by jacksonwilliams**
## Changes Proposed:
-  Update reagent store / pouch to be used account-wide

## Tests Performed:

- New account created.
- New character created.
- character teleported to bootybay
- added 1000g to character
- added 100 silverleaf to character
- added 100 peacebloom to character
- purchased reagent bag
- herbs deposited into reagent bag
- verified herbs visible in reagent bag

- migration sql run on database.
- server restarted

- new character created
- character teleported to bootybay
- interact with winzig
- check herb tab of reagent bank
- confirmed 100 silverleaf and 100 peacebloom was visable.

## How to Test the Changes:

1. goto winzig on character 1 / purchase reagent pouch
2. deposit items into the reagent bank
3. can items be seen on character 2

## Known Issues and TODO List:

- None

### CODE REVIEW

**Comment by jarrodjeffi** [comment]
We'd need some sort of migration script from `character_id` to `account_id` but this looks good.

File: modules/mod-winzig/src/WinzigCreature.cpp [hunk 1]
```diff
@@ -37,8 +37,9 @@ std::string npc_winzig::GetItemIcon(uint32 entry, uint32 width, uint32 height, i
 void npc_winzig::WithdrawItem(Player* player, uint32 entry)
 {
     // This query can be changed to async to improve performance, but there will be some visual bugs because the query will not be done executing when the menu refreshes
-    std::string query = "SELECT amount FROM reagent_bank WHERE character_id = " + std::to_string(player->GetGUID().GetCounter()) + " AND item_entry = " + std::to_string(entry);
-    QueryResult result = CharacterDatabase.Query("SELECT amount FROM reagent_bank WHERE character_id = " + std::to_string(player->GetGUID().GetCounter()) + " AND item_entry = " + std::to_string(entry));
+    // update to be account bound: player->GetSession()->GetAccountId()
```
Comment (jarrodjeffi) [inline modules/mod-winzig/src/WinzigCreature.cpp:40]: This comment is not necessary.

**Comment by jacksonwilliams** [comment]
> We'd need some sort of migration script from `character_id` to `account_id` but this looks good.

I've written up an SQL file that should migrate the data, but I would like to test it before I commit it.

File: modules/mod-winzig/src/WinzigCreature.cpp (hunk 1 above)
Comment (jacksonwilliams) [reply modules/mod-winzig/src/WinzigCreature.cpp]: removed.

**Comment by jacksonwilliams** [comment]
Tested.

New account created.
New character created.
character teleported to bootybay
added 1000g to character
added 100 silverleaf to character
added 100 peacebloom  to character

purchased reagent bag
herbs deposited into reagent bag

migration sql run on database.
server restarted

new character created
character teleported to bootybay
interact with winzig
check herb tab of reagent bank
confirmed 100 silverleaf and 100 peacebloom was visable.

File: modules/mod-winzig/src/WinzigCreature.cpp [hunk 2]
```diff
@@ -177,6 +177,7 @@ bool npc_winzig::OnGossipHello(Player* player, Creature* creature)
     AddGossipItemFor(player, GOSSIP_ICON_MONEY_BAG, GetItemIcon(2604, 30, 30, -18, 0) + "Other Trade Goods", ITEM_SUBCLASS_TRADE_GOODS_OTHER, 0);
     AddGossipItemFor(player, GOSSIP_ICON_MONEY_BAG, GetItemIcon(38682, 30, 30, -18, 0) + "Armor Vellum", ITEM_SUBCLASS_ARMOR_ENCHANTMENT, 0);
     AddGossipItemFor(player, GOSSIP_ICON_MONEY_BAG, GetItemIcon(39349, 30, 30, -18, 0) + "Weapon Vellum", ITEM_SUBCLASS_WEAPON_ENCHANTMENT, 0);
+    AddGossipItemFor(player, GOSSIP_ICON_MONEY_BAG, GetItemIcon(2724, 30, 30, -18, 0) + "Quest Items", ITEM_SUBCLASS_QUEST, 0);
```
Comment (jarrodjeffi) [inline modules/mod-winzig/src/WinzigCreature.cpp:180]: Did you add this?

Comment (jacksonwilliams) [reply modules/mod-winzig/src/WinzigCreature.cpp]: removed.

**Review by jarrodjeffi — APPROVED** [review]

**Pull request merged**